### DISCUSSION TITLE
Checkbox should popup a floating selector to pick states from instead of cycling through

### DISCUSSION THREAD

**Discussion by DeWarner** [iSoron/uhabits#1207] [Feature Requests]
instead of cycling through the checkbox states i think it would be easier if it just popped out with the options, highlighting the selected box

**Comment by end-ichi** [comment]
Agreed, I imagine something like this:
- long-press on the checkbox pops up a floating circle menu, [as seen in games like LoL](https://s.sk-gaming.com/image/image/3878fcee83a02eb9m.jpg). Possibly a small vibration happens. The user continues to press down.
- the quadrants are the options  (Yes, No, Skip, No Data), the centre is "cancel"
- the user swipes to the desired option and releases. Possibly another small vibration happens.

There will be problems when the entry the user tries to change is at the edge of the screen. In this case I have two ideas:
1. the floating menu is as close to the edge as possible while not being cut off. This would mean the centre is not where the user long-pressed which may be undesirable.
2. the floating menu is now a half circle (or quarter in thee corner case) and the visible part of it is divided equally into the options. This may cause problems with hitting the desired option especially on small screens which may also be undesirable.

Maybe someone else has a better idea. I'd really like it though if I didn't have to press twice (like with a non-floating pop-up menu in the screen centre for example).
